# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/src/bindings.rs
# Commit: eac17a862c29
# Full Hash: eac17a862c29a52ae1a389c5e55c0a1cec0db7b6
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-01-11 09:48:29
# Regressor Bug: 1600472
# File Overlap Count: 2
# Description:
#   Bug 1600472 - Disable allowing sacrificing subpixel anti-aliasing for small screens. r=jrmuizel
#   
#   This patch only allows sacrificing subpixel anti-aliasing when the
#   screen size is larger than WUXGA, and when the force disable pref is not
#   set. In the future, we may also add disable this for high end GPUs.
# ==============================================================================

diff -r b6e5faa0693d -r eac17a862c29 gfx/webrender_bindings/src/bindings.rs
--- a/gfx/webrender_bindings/src/bindings.rs	Fri Jan 10 18:24:13 2020 +0000
+++ b/gfx/webrender_bindings/src/bindings.rs	Fri Jan 10 16:57:55 2020 +0000
@@ -104,7 +104,6 @@
 /// cbindgen:field-names=[mNamespace, mHandle]
 type WrColorRange = ColorRange;
 
-
 #[repr(C)]
 pub struct WrSpaceAndClip {
     space: WrSpatialId,
@@ -1825,6 +1824,16 @@
 }
 
 #[no_mangle]
+pub extern "C" fn wr_transaction_set_quality_settings(
+    txn: &mut Transaction,
+    allow_sacrificing_subpixel_aa: bool
+) {
+    txn.set_quality_settings(QualitySettings {
+        allow_sacrificing_subpixel_aa
+    });
+}
+
+#[no_mangle]
 pub extern "C" fn wr_resource_updates_add_image(
     txn: &mut Transaction,
     image_key: WrImageKey,